### ISSUE TITLE
Allow users to set the word wrap width of the editor.

### ISSUE THREAD

**Issue by yougotwill** [marktext/marktext#1053]
### Describe your feature request
Allow users to set the word wrap width of the editor.

### What problem does this feature solve? [optional]
When trying to read markdown files sometimes:
1. The text feels very close together.
2. I want the document to take up more space on a bigger screen when reading. 

By increasing the word wrap width to say 120 the document becomes more readable.

e.g. in the user's Preference file we could have the following
```
{
...
"wordWrapWidth": 120,
...
}
```

### Screenshot Example
![screenshot](https://user-images.githubusercontent.com/14887287/58532754-716cd400-8221-11e9-8678-2dba08db8917.png)

**Comment by fxha** [comment]
Please also see #750 comments.

**Comment by Jocs** [comment]
@yougotwill What do you think is the appropriate width of the editing area? Or what range is appropriate？

**Comment by yougotwill** [comment]
@Jocs 
Taking from what Sublime Text does. My suggestion using the `wordWrapWidth` property mentioned above (maybe just call it `wrapWidth`?) would be:

**Default** should be the current value of 700px.
Basically if `wrapWidth` isn't explicitly set in the user preferences keep the default value we already have. This way we don't break the current experience for the users.

**wrapWidth = 0**
wraps to the width of the window (maybe with a little padding on each side).
basically width equals 100%

**0 > wrapWidth> 2560** 
Ideally I would say let the user set any value greater than 0 but if you want a limit then use 4K which is 2560px.

Of course if width set is greater than the app window size then we would need horizontal scrolling.

Feel free to ask for any clarification. :+1:

**Comment by Jocs** [comment]
> Of course if width set is greater than the app window size then we would need horizontal scrolling.

I think the `wrapWidth` should be the min value of app window width(not include side bar) and the preference setting value.

**Comment by fxha** [comment]
I think the easisest would be to allow to set `max-width` of `ch`, `px` or `%` of the editor area and only validate these types but not the values. I tested it and it works great, invalid values would fill the complete screen but don't produce any issues.

**Comment by yougotwill** [comment]
@fxha I looked at your PR. Nice solution!
